thirdeye/thirdeye-frontend/tests/integration/pods/components/range-pill-selectors/component-test.js

```diff
@@ -0,0 +1,121 @@
+import { later } from "@ember/runloop";
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { setUpTimeRangeOptions } from 'thirdeye-frontend/utils/manage-alert-utils';
+
+module('Integration | Component | range pill selectors', function(hooks) {
+  setupRenderingTest(hooks);
+
+  const PILL_CLASS = '.te-pill-selectors';
+  const MAXTIME = 1520863199999; // tells us how much data is available for the selected metric
+  const START_DATE = 1513151999999; // arbitrary start date in milliseconds
+  const END_DATE = 1520873345292; // arbitrary end date in milliseconds
+  const TIME_PICKER_INCREMENT = 5; // tells date picker hours field how granularly to display time
+  const ACTIVE_DURATION = '1m'; // setting this date range option as default
+  const UI_DATE_FORMAT = 'MMM D, YYYY hh:mm a'; // format for date picker to use (usually varies by route or metric)
+  const DISPLAY_DATE_FORMAT = 'YYYY-MM-DD HH:mm'; // format used consistently across app to display custom date range
+  const TODAY = moment().startOf('day').add(1, 'days');
+  const TWO_WEEKS_AGO = moment().subtract(13, 'days').startOf('day');
+  const ACTIVE_RANGE_END = moment(END_DATE).format(DISPLAY_DATE_FORMAT);
+  const ACTIVE_RANGE_START = moment(START_DATE).format(DISPLAY_DATE_FORMAT);
+  const TIME_RANGE_OPTIONS = setUpTimeRangeOptions(['1m', '3m'], ACTIVE_DURATION); // using helper to generate these
+  const PRESET_RANGES = {
+    'Today': [moment(), TODAY],
+    'Last 2 weeks': [TWO_WEEKS_AGO, TODAY]
+  };
+
+  test('Confirming that range-pill-selector component renders and dates are selected properly', async function(assert) {
+    // setUpTimeRangeOptions will generate an array such as this
+    // [ { name: "3 Months", value: "3m", start: Moment, isActive: true },
+    // { name: "Custom", value: "custom", start: null, isActive: false } ]
+
+    this.set('onRangeSelection', (actual) => {
+      let expected = {
+        start: TWO_WEEKS_AGO,
+        end: TODAY
+      };
+      assert.deepEqual(actual, expected, 'selected start/end dates are passed to external action');
+    });
+
+    // Template block usage:
+    await render(hbs`
+      {{range-pill-selectors
+        title="Testing range pills"
+        maxTime=MAXTIME
+        uiDateFormat=UI_DATE_FORMAT
+        activeRangeEnd=ACTIVE_RANGE_END
+        activeRangeStart=ACTIVE_RANGE_START
+        timeRangeOptions=TIME_RANGE_OPTIONS
+        timePickerIncrement=TIME_PICKER_INCREMENT
+        predefinedRanges=PRESET_RANGES
+      }}
+    `);
+
+    const $rangePill = this.$(`${PILL_CLASS}__item`);
+    const $rangeTitle = this.$(`${PILL_CLASS}__title`);
+    const $rangePickerModal = this.$('.daterangepicker');
+    const $rangeInput = this.$('.daterangepicker-input');
+    const $rangePresets = this.$('.daterangepicker .ranges ul li');
+    const $customPill = this.$(`${PILL_CLASS}__item[data-value="custom"]`);
```

Comment:
fyi, you can do  `$rangePill.find()` so you don't need to grab the item again from the DOM